geode-redis/src/main/java/org/apache/geode/redis/internal/executor/hash/RedisHashCommands.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.executor.hash;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.geode.redis.internal.ByteArrayWrapper;
+
+public interface RedisHashCommands {
+  int hset(ByteArrayWrapper key, List<ByteArrayWrapper> fieldsToSet, boolean NX);
+
+  int hdel(ByteArrayWrapper key, List<ByteArrayWrapper> fieldsToRemove);
+
+  // TODO: refactor this onto something like RedisKeyCommands
+  boolean del(ByteArrayWrapper key);
+
+  Collection<ByteArrayWrapper> hgetall(ByteArrayWrapper key);
```

Comment:
i think it would be nice if hgetall returned a collection of enttries.  That has more semantic meaning, but I understand that the caller would end up just flattening it to send it over the wire.

Reply:
And it would not be serializable

Reply:
I do think it makes more sense for hgetall to return a Collection<ByteArrayWrapper> instead of Collection<Entry<ByteArrayWrapper, ByteArrayWrapper>> because the output of the redis HGETALL command is a flat list. The output is "key1 value1 key2 value2". It does not have structure (for example "{key1 value1} {key2 value2}". So by having hgetall return a simple Collection it actually corresponds to the public REDIS HGETALL output.